packages/cardpay-sdk/sdk/reward-manager/base.ts

```diff
@@ -496,6 +500,174 @@ export default class RewardManager {
+    let rewardSafeOwner = await rewardManager.getRewardSafeOwner(safeAddress);
+
+    let token = new this.layer2Web3.eth.Contract(ERC20ABI as AbiItem[], tokenAddress);
+    let safeBalance = new BN(await token.methods.balanceOf(safeAddress).call());
+
+    if (!(rewardSafeOwner == from)) {
+      throw new Error(
+        `Reward safe owner is NOT the signer of transaction.
+The owner of reward safe ${safeAddress} is ${rewardSafeOwner}, but the signer is ${from}`
+      );
+    }
+    let weiAmount = amount ? new BN(toWei(amount)) : safeBalance;
+    if (weiAmount.gt(safeBalance)) {
+      throw new Error(`Insufficient funds for inside reward safe`);
+    }
+
+    let transfer = await token.methods.transfer(rewardSafeOwner, weiAmount);
+    let transferPayload = transfer.encodeABI();
+
+    let { nonce, onNonce, onTxnHash } = txnOptions ?? {};
+    let transferNonce: BN;
+    if (nonce != null) {
+      // a passed in nonce represents the next nonce to use, so we add 1 to it
+      // to get the nonce we'd want to use for the transfer execTransaction
+      transferNonce = nonce.add(new BN(1));
+    } else {
+      let rewardSafe = new this.layer2Web3.eth.Contract(GnosisSafeABI as AbiItem[], safeAddress);
+      let currentNonce = new BN(await rewardSafe.methods.nonce().call());
+      transferNonce = currentNonce.add(new BN('1'));
+    }
+    // to token contract
+    let innerEstimate = await gasEstimate(
+      this.layer2Web3,
+      safeAddress,
+      tokenAddress,
+      '0',
+      transferPayload,
+      0,
+      tokenAddress
+    );
+
+    let fullSignatureInnerExec = await fullSignatureTxAsBytes(
+      this.layer2Web3,
+      tokenAddress,
+      0,
+      transferPayload,
+      0,
+      innerEstimate,
+      safeAddress, //or maybe zero address
+      transferNonce,
+      rewardSafeOwner,
+      safeAddress,
+      rewardManagerAddress
+    );
+
+    let gasCost = new BN(innerEstimate.dataGas).add(new BN(innerEstimate.baseGas)).mul(new BN(innerEstimate.gasPrice));
+    if (weiAmount.lt(gasCost)) {
+      throw new Error(
+        `Reward safe does not have enough to pay for gas when claiming rewards. The reward safe ${safeAddress} balance for token ${tokenAddress} is ${fromWei(
+          safeBalance
+        )}, amount being claimed is ${amount}, the gas cost is ${fromWei(gasCost)}`
+      );
+    }
+    let payload = (await this.getRewardManager()).methods
+      .withdrawFromRewardSafe(
+        tokenAddress,
+        weiAmount,
+        innerEstimate.safeTxGas,
+        innerEstimate.baseGas,
+        innerEstimate.gasPrice,
+        innerEstimate.gasToken,
+        fullSignatureInnerExec
+      )
+      .encodeABI();
+
+    if (nonce == null) {
+      nonce = getNextNonceFromEstimate(innerEstimate);
+      if (typeof onNonce === 'function') {
+        onNonce(nonce);
+      }
+    }
+    let estimate = await gasEstimate(this.layer2Web3, safeAddress, rewardManagerAddress, '0', payload, 0, tokenAddress);
```

Comment:
`CannotEstimate` error  here. When I just hardcode the gas, I get safe transaction reverted.